[next] Timestamp - Reader can see relative time of when that comment was posted

okbel opened the pull request:
### RelativeTime - Reader can see relative time of when that comment was posted
- Adds RelativeTime component
- Adds Timestamp component
- Adds "createdAt" field to the comment schema/resolvers
- Translations
- Show meaningful css names in dev
- Turn on antialiasing for font rendering
- Tests and test support for MatchMedia uses

alinhle committed:
Do not rerun relay compiler on its own changes

alinhle commented on .vscode/settings.json:
```diff
@@ -16,5 +16,6 @@
   "tslint.jsEnable": true,
-  "typescript.tsdk": "node_modules/typescript/lib"
+  "typescript.tsdk": "node_modules/typescript/lib",
+  "git.ignoreLimitWarning": true
```
Can you turn it to false and see if you still have the issue?
Just making sure we really need this, it seems to make sense to keep the default.

alinhle commented on src/core/client/framework/lib/bootstrap/TalkContext.tsx:
```diff
@@ -18,7 +23,7 @@ const { Provider, Consumer } = React.createContext<TalkContext>({} as any);
 export const TalkContextConsumer = Consumer;
 
-/**
+/**s
```
Here is a lonely `s`

alinhle commented on src/core/client/stream/components/Comment.spec.tsx:
```diff
@@ -8,6 +8,7 @@ it("renders username and body", () => {
       username: "Marvin",
     },
     body: "Woof",
+    createdAt: new Date().toISOString(),
```
This won't work well with jest snapshot, adding a fixed time would be better.

alinhle commented on src/core/client/stream/components/Comment.spec.tsx:
```diff
@@ -20,6 +21,7 @@ it("renders with gutterBottom", () => {
       username: "Marvin",
     },
     body: "Woof",
+    createdAt: new Date().toISOString(),
```
See above

alinhle commented on src/core/client/framework/lib/bootstrap/createContext.tsx:
```diff
@@ -16,6 +19,19 @@ interface CreateContextArguments {
   init?: ((context: TalkContext) => void | Promise<void>);
 }
 
```
We should add a comment here.

alinhle commented on src/core/client/ui/components/Timestamp/Timestamp.spec.tsx:
```diff
@@ -0,0 +1,19 @@
+  };
+  const tree = create(<Timestamp {...props} />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
As suggested before, a test to check if consuming from the context is working correctly, would be helpful.

alinhle commented on src/core/client/ui/components/Timestamp/Timestamp.tsx:
```diff
@@ -0,0 +1,37 @@
+  date: string;
+  live?: boolean;
+  classes: typeof styles;
+  formatter?: Formatter;
```
This is never used, shouldn't it be passed to `TimeAgo`?

okbel replied:
This is old. Removed!

alinhle commented on src/locales/de/framework.ftl:
```diff
@@ -3,3 +3,37 @@
 
 framework-validation-required = Dies ist ein Pflichtpfeld.
+
+framework-timeago =
```
These German Translations are in Spanish?

okbel replied:
These are for you haha. I'm really rusty with german.

alinhle replied:
Haha sure

alinhle commented on src/locales/en-US/framework.ftl:
```diff
@@ -3,3 +3,38 @@
 framework-validation-required = This field is required.
+
+framework-timeago =
+  { $value ->
```
The `Spanish` translations is using `en 1 hora` but the English translation `1 hour from now`, I think it's better when we stay consistent and also translate to `in 1 hour`.

okbel replied:
💅 : ok

alinhle replied:
I tried that the last time actually and I first didn't know how to do it, but now I do, we can talk about it on Monday too :-)

alinhle replied:
Actually in our current Talk we also changed from `1 hour from now` to `in 1 hour`, I don't know any reason who would use `1 hour from now`.

alinhle commented on src/core/client/ui/components/Timestamp/Timestamp.tsx:
```diff
@@ -0,0 +1,37 @@
+import * as styles from "./Timestamp.css";
+
+interface InnerProps {
+  date: string;
```
We should add support for `className` here.

okbel replied:
👍

alinhle commented on src/core/client/ui/components/Timestamp/Timestamp.css:
```diff
@@ -0,0 +1,4 @@
+.root {
+  composes: button from "talk-ui/shared/typography.css";
```
We need to add typography styling from Sam or if already available from the Stylesheet

okbel replied:
👍

alinhle commented on src/core/client/stream/components/Comment.tsx:
```diff
@@ -24,6 +25,7 @@ const Comment: StatelessComponent<CommentProps> = props => {
       <Typography className={styles.author} gutterBottom>
         {props.author && props.author.username}
       </Typography>
+      <Timestamp date={props.createdAt} />
```
We need to figure out a way to have `responsiveness` here.

alinhle replied:
I'm having a reusable component in mind, that we can always rely on.

alinhle replied:
I'll work with you on this, I'm gonna test if one of my components work that I wrote in one of my earlier projects.

okbel replied:
👍

alinhle commented:
Also the large introspection files are still in the commit history, can you take another look?

okbel committed:
Implement Timestamp

Merge branch 'next' into timestamp

Fix test and translations

Merge branch 'timestamp' of github.com:coralproject/talk into timestamp

* 'timestamp' of github.com:coralproject/talk:
  Fix test and translations
  Implement Timestamp

okbel committed:
Merge branch 'next' of github.com:coralproject/talk into timestamp

* 'next' of github.com:coralproject/talk:
  Adapt files.exclude (#1736)
  [next] Jest implementation for React Components (#1733)
  [next] Remove nodemon (#1725)
  Use JSDocs comments (#1727)
  feat: added support for --only flag
  Do not rerun relay compiler on its own changes

alinhle committed:
Update snapshots

alinhle committed:
Merge branch 'next' into timestamp

alinhle committed:
Rename Timestamp to RelativeTime

alinhle committed:
Show meaningful css names in dev

alinhle committed:
Make design responsive

alinhle committed:
Turn on better font rendering

alinhle committed:
Organize

alinhle committed:
Update snapshts

alinhle committed:
Add tests

alinhle committed:
Fix linting

alinhle committed:
Update tests

alinhle committed:
Remove unnused value

alinhle committed:
Fix remaining test

alinhle committed:
Typography support error color

wyattjoh commented on package.json:
```diff
@@ -45,6 +45,7 @@
     "mongodb": "^3.0.10",
     "passport": "^0.4.0",
     "performance-now": "^2.1.0",
+    "react-timeago": "^4.1.9",
```
Should be in devDep instead of dep

alinhle replied:
👍

wyattjoh commented on src/core/client/framework/lib/bootstrap/TalkContext.tsx:
```diff
@@ -1,14 +1,19 @@
   localeMessages: MessageContext[];
+
+  // formatter for timeago.
+  timeagoFormatter: Formatter | null;
```
Why not use:

```typescript
timeagoFormatter?: Formatter;
```

instead?

alinhle replied:
I didn't want this to be optional, but to explicitly set to null in special cases e.g. for testing. Normally this is required.

alinhle replied:
Thought about it again and made it optional :-P

wyattjoh committed:
Merge branch 'next' into timestamp

alinhle committed:
Move react-timeago to dev

alinhle committed:
Merge branch 'timestamp' of ssh://github.com/coralproject/talk into timestamp

alinhle committed:
Merge branch 'next' into timestamp

alinhle committed:
Make timeagoFormatter optional
